subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaBasePlugin.java

```diff
@@ -129,6 +131,11 @@ public void apply(final Project project) {
         configureBuildNeeded(project);
         configureBuildDependents(project);
         bridgeToSoftwareModelIfNecessary(projectInternal);
+        configureToolchainsTask(project);
```

Comment:
:x: I feel like this isn't the right location for this task. You may want to list Java toolchains even on non Java projects. the `jvm-ecosystem` plugin feels like a better place, but I'm wondering if this shouldn't just be a top-level task created just like `help` and `outgoingVariants`.

Reply:
These are actually created [exactly like those tasks](https://github.com/gradle/gradle/blob/master/subprojects/diagnostics/src/main/java/org/gradle/api/plugins/HelpTasksPlugin.java#L66). The reason I didn't add this task to the `diagnostics` plugin is to avoid a dependeny from `diagnostics` to `platform-jvm`. We may actually be able to move it over once the probe services is refactored and lives in `jvm-services`. Or I just move it to `jvm-ecosystem` and that's good enough. I thought we want to move away from putting everything into the `plugins` module. WDYT?

Reply:
As discussed, leaving it in here but now auto-applying so it can stay where it is without new dependencies

Reply:
What do you mean? What plugin is auto applied? I still think that the Java base plugin shouldn't configure this task, it should be something else.

Reply:
You have to wait for the new commits to be pushed :) As we discussed, we now do it the same way the help task does it. A new plugin is auto-applied which will register the task on the project. We don't need the change in `JavaBasePlugin` anymore.